docs/src/modules/components/DemoPropsForm.tsx

```diff
@@ -131,26 +146,31 @@ function ControlledColorRadio(props: any) {
   );
 }
 
-export default function ChartDemoPropsForm<T extends { [k: string]: any } = {}>({
+export default function ChartDemoPropsForm<T extends string>({
   componentName,
   data,
+  props,
   onPropsChange,
 }: ChartDemoPropsFormProps<T>) {
-  const initialProps = {} as { [k in keyof T]: any };
-  let demoProps = {} as { [k in keyof T]: any };
-
-  data.forEach((p) => {
-    demoProps[p.propName] = p.defaultValue;
-
-    initialProps[p.propName] = p.defaultValue;
-  });
-  const [props, setProps] = React.useState<T>(initialProps as T);
+  const initialProps = React.useRef<Record<T, any>>(
+    data.reduce(
+      (acc, { propName, defaultValue }) => {
+        acc[propName] = defaultValue;
+        return acc;
+      },
+      {} as Record<T, any>,
+    ),
+  );
 
   React.useEffect(() => {
-    onPropsChange(props);
-  }, [props, onPropsChange]);
-
-  demoProps = { ...demoProps, ...props };
+    data.reduce(
+      (acc, { propName, defaultValue }) => {
+        acc[propName] = defaultValue;
+        return acc;
+      },
+      {} as Record<T, any>,
+    );
+  }, [data]);
```

Comment:
Now that the props are controlled from the parent, what about saving the initial props with a `useMemo`, or save the reset function with `const resetProps = React.useCallback(() => {...}, [data])`?

Reply:
Used `useMemo` instead because we also need a ref of the object to check for equality in case it changed